app/controllers/members_controller.rb

```diff
@@ -0,0 +1,14 @@
+class MembersController < ApplicationController
+
+  def create
+    event = Event.find(params[:event_id])
+    event.members << current_user
+    redirect_to event_path(event)
+  end
+
+  def delete
+    event = Event.find(params[:event_id])
+    event.member = current_user
```

Comment:
From what I know you do not have `member` method only `members` is your method.  Since you would like to remove current user from event first you need to check if you can do that (check the ability -cancan). Second when you do action delete you should really remove relation between user and event to do that use [documentation](https://guides.rubyonrails.org/association_basics.html#has-and-belongs-to-many-association-reference)

Reply:
This is still valid comment.

Reply:
I`m trying to do it according to this:
https://guides.rubyonrails.org/association_basics.html#has-and-belongs-to-many-association-reference

I want to create callbacks - notices. 
First I have in Event Model 
  has_and_belongs_to_many :members, class_name: "User", before_remove: notice_before_remove

and in controller event in private:
```
def notice_before_remove(event)
      if @event.has_member?
        flash[:notice] = "There are some members in your event. Are you sure that you want to remove it?"
      else
        flash[:notice] = "Are you sure?"
      end
    end
```

I got an error NameError in EventsController#index
undefined local variable or method `notice_before_remove' for #<Class:0x00005647608030d8>

Where is my mistake?

Reply:
First you don't want to do that on Model level at that point. To do it with easier approach for now do this validation check on controller level not in model. 
Second if you put in model `before_remove: notice_before_remove` and in controller you have `notice_before_remove` it will not work. These are 2 different classes and you cannot share code between them.